syntax = "proto3";

service PeerService {
  // Cargar archivo
  rpc UploadFile (FileRequest) returns (FileResponse);

  // Descargar un archivo en chunks de datos
  rpc DownloadFile (FileRequest) returns (stream FileChunk);
}

message FileRequest {
  string filename = 1; 
}

message FileResponse {
  bool success = 1;     // Indica si la operación fue exitosa
  string message = 2;   // Mensaje con información adicional
}

message FileChunk {
  bytes content = 1;    // Fragmento de datos del archivo
}
